WIP: chore: Upgrade Argo to v3.4.16

gmfrasca opened the pull request:
**Description of your changes:**
Upgrade backend to Argo v3.4.16

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 
<!--
   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.
-->

gmfrasca commented on backend/src/apiserver/client/sql.go:
```diff
@@ -20,6 +20,10 @@ import (
 )
 
+type postgresConfig struct {
+	DSN string
```
so this right now is very trivial, but there is no programtic config util for postgres like there is in the default mysql driver.  With that said, i wonder if it makes sense to essentially emulate the mysql Config and populate the config fields and provide the dsn dynamically via a formatDSN() method

rimolive replied:
If we want to make the DSN format universal among database vendors, it makes sense. My concern is that this would introduce unnecessary bugs

gmfrasca committed:
pull argo v3.4.16 upstream

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

gmfrasca committed:
upgrade to Argo v3.4.16

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

gmfrasca committed:
Update ValidateWorkflow calls

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

droctothorpe committed:
feat(kubernetes_platform): Update kubernetes_platform go package to i… (#10442)

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: edmondop <edmondo.porcu@gmail.com>
Co-authored-by: tarat44 <32471142+tarat44@users.noreply.github.com>

connor-mccarthy committed:
docs(components): internal

PiperOrigin-RevId: 607802006

connor-mccarthy committed:
chore(components): bump highest support KFP SDK version in GCPC to KFP SDK 2.7.0

PiperOrigin-RevId: 608646229

hbelmiro committed:
fix(backend): fixes "cannot save parameter" error message. Fixes #9678 (#10459)

Signed-off-by: hbelmiro <helber.belmiro@gmail.com>

Commit:
fix(rlhf): Supporting adapter only output for reward model training

PiperOrigin-RevId: 608740017

Commit:
chore(components): internal

PiperOrigin-RevId: 609081795

Commit:
docs(components): internal

PiperOrigin-RevId: 609107204

Commit:
feat(components): Added experimental args to batch_prediction_pairwise component

PiperOrigin-RevId: 609129336

Commit:
chore(components): release GCPC SDK 2.10.0

PiperOrigin-RevId: 609459972

droctothorpe committed:
feat(Backend + SDK): Update kfp backend and kubernetes sdk to support tolerations (#10471)

* feat(Backend + SDK): Update kfp backend and kubernetes sdk to support tolerations

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: edmondop <edmondo.porcu@gmail.com>
Co-authored-by: tarat44 <32471142+tarat44@users.noreply.github.com>

* Address PR review 1

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: edmondop <edmondo.porcu@gmail.com>
Co-authored-by: tarat44 <32471142+tarat44@users.noreply.github.com>

* Refactor add_toleration to use Python primitives

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: edmondop <edmondo.porcu@gmail.com>
Co-authored-by: tarat44 <32471142+tarat44@users.noreply.github.com>

* Update go.mod to pull in latest kubernetes_platform package

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: edmondop <edmondo.porcu@gmail.com>
Co-authored-by: tarat44 <32471142+tarat44@users.noreply.github.com>

---------

Signed-off-by: droctothorpe <mythicalsunlight@gmail.com>
Co-authored-by: edmondop <edmondo.porcu@gmail.com>
Co-authored-by: tarat44 <32471142+tarat44@users.noreply.github.com>

Commit:
feat(components): Add configurable image prefix to llm utility method

PiperOrigin-RevId: 609560776

Commit:
chore(components): internal change

PiperOrigin-RevId: 609757226

Commit:
feat(components): Add CMEK support to `preview.llm.rlhf_pipeline`

PiperOrigin-RevId: 609832020

roytman committed:
feat(Backend + SDK): Update kfp backend and kubernetes sdk to support ConfigMaps as volumes and as env variables (#10483)

* Update kfp backend and kubernetes sdk to support ConfigMaps as volumes and as env

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* update go.mod, apiserver.csv and driver.csv

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* add test/snapshot/data files

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* fix tests

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* go mod tidy

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* update backend/third_party_licenses/apiserver.csv

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* update backend/third_party_licenses/apiserver.csv

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* fix comments

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* fix comments

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

* update go.mod, apiserver.csv and driver.csv

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

---------

Signed-off-by: Alexey Roytman <roytman@il.ibm.com>

Commit:
feat(components): Release Forecasting training pipelines to V1 namespace

PiperOrigin-RevId: 610830518

KevinGrantLee committed:
fix(sdk): Prevents dsl.ParallelFor over single parameter from compiling. (#10494)

* fix(sdk): Prevents dsl.ParallelFor over single paramter from compiling.

* fix(sdk): Prevents dsl.ParallelFor over single paramter from compiling.

* update PR number in release notes

* formatting

* Add compiler_test.py test for single param compile failure

* Update some docstrings and add todo

* formatting

* Update sdk/python/kfp/compiler/compiler_test.py

Co-authored-by: Connor McCarthy <mccarthy.connor.james@gmail.com>

* Update sdk/python/kfp/compiler/compiler_test.py

Co-authored-by: Connor McCarthy <mccarthy.connor.james@gmail.com>

* Update sdk/python/kfp/dsl/for_loop.py

Co-authored-by: Connor McCarthy <mccarthy.connor.james@gmail.com>

* Use print_and_return and other small changes

* typo

* typo

---------

Co-authored-by: Connor McCarthy <mccarthy.connor.james@gmail.com>

Commit:
fix(components): Pass tuned model checkpoint to inference pipeline after RLHF tuning

PiperOrigin-RevId: 610918020

KevinGrantLee committed:
test: Add ParallelFor compile test over single artifact. (#10531)

* .

* formatting

Commit:
fix(components): rename custom task calibration_score_rubric -> score_rubric

PiperOrigin-RevId: 611161020

jsondai committed:
chore(components): Switch default machine type for LLM Text Generation Eval pipeline and components to e2-standard-4

PiperOrigin-RevId: 612531671

TheMichaelHu committed:
fix(components): Propagate location to sub-components in AutoSxS

PiperOrigin-RevId: 612553652

Tomcli committed:
feat(backend + SDK): Add backend and SDK support to use Kubernetes FieldPath as env (#10496)

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

Commit:
chore(components): Change docker image URI used by `preview.llm` pipelines

PiperOrigin-RevId: 612662160

hbelmiro committed:
fix(docs): Updated legal info due to migration from CLA to DCO (#10501)

* Updated legal info due to migration from CLA to DCO

Signed-off-by: hbelmiro <helber.belmiro@gmail.com>

* Updated DCO link

Signed-off-by: hbelmiro <helber.belmiro@gmail.com>

---------

Signed-off-by: hbelmiro <helber.belmiro@gmail.com>

Tomcli committed:
feat(backend + SDK): Add Backend and SDK support for timeout in pod spec (#10481)

* Add backend and sdk support for pod spec timeout

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

* fix conflicts

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

---------

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

Commit:
fix(components): Return None as sliced feature attribution values for the classes which are not predicted in bp outputs

PiperOrigin-RevId: 612920651

Commit:
feat(components): Add CMEK validation to `preview.llm.infer_pipeline`

PiperOrigin-RevId: 612956960

revit13 committed:
feat(Backend + SDK): Update kfp backend and kubernetes sdk to support ImagePullPolicy (#10417)

* feat(Backend + SDK): Update kfp backend and kubernetes sdk to support ImagePullPolicy.

Signed-off-by: Revital Sur <eres@il.ibm.com>

* Fix format.

Signed-off-by: Revital Sur <eres@il.ibm.com>

* Update apiserver.csv.

Signed-off-by: Revital Sur <eres@il.ibm.com>

* Update licenses.

Signed-off-by: Revital Sur <eres@il.ibm.com>

---------

Signed-off-by: Revital Sur <eres@il.ibm.com>

connor-mccarthy committed:
docs(components): fix `create_custom_training_job_from_component` docs rendering

PiperOrigin-RevId: 612985431

Commit:
feat(components): Implement the train time evaluation in reward model training. With the train time eval dataset available, the pipeline outputs the accuracy and cross entropy metrics to the log

PiperOrigin-RevId: 613057150

PratyushaRavi committed:
fix(samples): Updated samples/core to V2 (#9879)

* Updated output_a_directory.py to V2

* Update output_a_directory_test.py to V2

* Update parallel_join.py to V2

* Update multiple_outputs.ipynb to V2

* Update multiple_outputs_test.py to V2

* Updated kfp_env_validation to V2

* Updated loop_parallelism to V2

Commit:
chore(components): Consolidate validation components in `preview.llm.rlhf_pipeline`

PiperOrigin-RevId: 613366137

Tomcli committed:
feat(backend): Upgrade go version to 1.20 (#10502)

* upgrade go version to 1.21

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

* upgrade integration test to go 1.21

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

* refresh go mod tidy

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

* fix license

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

* update go-sqlite3 to v1.14.19 to support go 1.21 on cache server

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

* downgrade go version to 1.20 and revert dockerfile upgrade

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

---------

Signed-off-by: Tommy Li <Tommy.chaoping.li@ibm.com>

Commit:
docs(components): Modify the GetModel documentation

PiperOrigin-RevId: 614771557

Commit:
chore(components): Update AutoSxS and RLHF image tags

PiperOrigin-RevId: 614785091

champon1020 committed:
fix(backend): Fixes response status of http error code when uploading duplicate pipeline [Fixes  #10311] (#10546)

Validate the error code of pipeline creation in order to return
the status conflict when the error represents AlreadyExists.

Signed-off-by: champon1020 <nagatelu1020@gmail.com>

Commit:
chore: Change stalebot rules (#10547)

daysUntilClose is set to 90 days. This is too much for an issue to keep opened as there is daysUntilStale set to 90 days.
That would keep an issue opened for 6 months, and now there are +600 issues opened in the repository.

We need to start working on keeping the repository healthy.

Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>

revit13 committed:
fix(kubernetes_platform): Add optional field to SecretAsVolume and ConfigMapAsVolume. Fixes #10548 (#10549)

* fix(kubernetes_platform): Add optional field to SecretAsVolume and ConfigMapAsVolume.

Signed-off-by: Revital Sur <eres@il.ibm.com>

* Update comment.

Signed-off-by: Revital Sur <eres@il.ibm.com>

---------

Signed-off-by: Revital Sur <eres@il.ibm.com>

gmfrasca committed:
Add NodeStatus pod name retrieval function

- Argo 3.4.16 upgrade introduces a breaking change with inconsistent node.ID vs
  node.Name
- introduce a function in workflow.go to conditionally handle this

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

TheMichaelHu committed:
feat(components): Add CMEK support to AutoSxS pipeline

PiperOrigin-RevId: 615498240

Commit:
feat(components): Update _LLM_EVAL_VERSION to v0.6

PiperOrigin-RevId: 615562899

gmfrasca committed:
Remove PNS Executor manifests and containerRuntimeExecutor ConfigMap Key

- PNS Executor was removed in Argo v3.4, so manifests no longer valid
- WorkflowController will fail to start if `containerRuntimeExecutor`
  provided as input parameter, so remove from WC ConfigMap and CM
  patches

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

jsondai committed:
chore(components): update container image of endpoint batch predict component for vulnerability patch

PiperOrigin-RevId: 615593420

chensun committed:
chore(backend): Update kfp driver and launcher images (#10561)

Signed-off-by: Chen Sun <chensun@users.noreply.github.com>

Commit:
fix(components): Add relevant component and pipeline inputs/outputs to support creating ModelEvaluations as part of the AutoSxS Metrics component

PiperOrigin-RevId: 615675169

Commit:
feat(components): Release Forecasting training pipelines to V1 namespace

PiperOrigin-RevId: 615914679

chensun committed:
chore(release): bumped version to 2.1.0

Commit:
feat(components): Add location validation to `preview.llm.rlhf_pipeline`

PiperOrigin-RevId: 616229944

chensun committed:
Revert "chore(release): bumped version to 2.1.0"

This reverts commit 4d90770dd319b7b342d601a3f04562f46301d583.

Commit:
chore(components): Testing fix

PiperOrigin-RevId: 616240307

junggil committed:
fix(samples): Update resource_spec, retry, secret samples to v2 pipelines (#9876)

* Update resource_spec, retry, secret samples to v2 pipelines

* Update resource_spec, retry, secret samples to v2 pipelines

amadhusu committed:
fix: Modified the comment/text for pipeline_version_id (#10581)

Signed-off-by: Achyut Madhusudan <amadhusu@redhat.com>

Commit:
chore(components): Update AutoSxS and RLHF image tags

PiperOrigin-RevId: 616895855

chensun committed:
chore: Update api-generator to use golang 1.20 (#10580)

Signed-off-by: Chen Sun <chensun@users.noreply.github.com>

connor-mccarthy committed:
fix(docs): make full version dropdown show on all KFP SDK docs versions (#10577)

Signed-off-by: connor-mccarthy <mccarthy.connor.james@gmail.com>

Commit:
feat(components): Implement new component to preprocess and validate inputs for rlhf

PiperOrigin-RevId: 617285265

Commit:
chore(components): GCPC 2.11.0 Release

PiperOrigin-RevId: 617287233

Commit:
fix(components): Fix missing pipeline parameters

PiperOrigin-RevId: 617579452

chensun committed:
chore: Update release image (#10599)

Signed-off-by: Chen Sun <chensun@users.noreply.github.com>

Commit:
fix(components): Remove the unused resolve_candidate_columns from function_based

PiperOrigin-RevId: 617802429

amadhusu committed:
fix: Modified the swagger json files according to the modified proto files. (#10591)

Signed-off-by: Achyut Madhusudan <amadhusu@redhat.com>

Commit:
feat(components): Copy text generation eval and text classification evak pipelines from preview to v1

PiperOrigin-RevId: 618017914

Commit:
docs(components): internal

PiperOrigin-RevId: 618313694

chensun committed:
chore(release): bumped version to 2.1.0

Commit:
feat(components): Report TensorBoard metrics for `preview.llm.rlhf_pipeline` in real time

PiperOrigin-RevId: 618924675

abaland committed:
feat(kubernetes_platform): Update kubernetes_platform go package to include generic ephemerl volume (#10602)

* feat(kubernetes_platform): Update kubernetes_platform go package to include Generic Ephemeral Volume resources
Signed-off-by: abaland <abaland@indeed.com>

* feat(kubernetes_platform): Update kubernetes_platform go package to include Generic Ephemeral Volume resources (include metadata)
Signed-off-by: abaland <abaland@indeed.com>

freefood89 committed:
fix(metadata envoy): upgrade envoy and config from 1.12 to 1.27 (#10589)

Signed-off-by: Rentaro Matsukata <freefood89@users.noreply.github.com>

Commit:
fix(components): Remove the unused resolve_data_paths from function_based

PiperOrigin-RevId: 619026115

Commit:
chore: Add Question issue template (#10557)

Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>

Commit:
feat(kubernetes_platform): Update kubernetes_platform go package to include node affinities and pod (anti)affinities (#10583)

* feat(kubernetes_platform): Update kubernetes_platform go package to include nodeaffinities and pod (anti)affinities

Signed-off-by: Cornelis Boon <cjidboon94@gmail.com>

* rename affinity objects and fields to match k8s spec semantics

Signed-off-by: Cornelis Boon <cjidboon94@gmail.com>

* rename *AffinityRule -> *AffinityTerm and add missing affinity data

Signed-off-by: Cornelis Boon <cjidboon94@gmail.com>

---------

Signed-off-by: Cornelis Boon <cjidboon94@gmail.com>

Commit:
chore(components): Add test machine spec support to `preview.llm` pipelines

PiperOrigin-RevId: 619378459

Commit:
fix(components): Ensure `preview.llm.rlhf_pipeline` runs if no `tensorboard_id` is provided

PiperOrigin-RevId: 619646459

Commit:
fix(components): Remove the unused functions from function_based

PiperOrigin-RevId: 619665186

Commit:
feat(components): Add model name preprocess component; Use publisher model if user uploaded model is non-tuned

PiperOrigin-RevId: 619714978

Commit:
feat(components): add task_type as a parameter to rlaif

PiperOrigin-RevId: 619744783

Commit:
feat(components): AutoSxS GA pending release

PiperOrigin-RevId: 619748191

Commit:
fix(components): Update service account comment

PiperOrigin-RevId: 619982662

connor-mccarthy committed:
chore(components): GCPC 2.12.0 Release

PiperOrigin-RevId: 619990671

Tomcli committed:
Update loop_output.py example for the new parallel loop type requirement (#10637)

Signed-off-by: tomcli <tommy.chaoping.li@ibm.com>

Commit:
feat(components): Added support for text-bison@002 to preview.llm.rlhf_pipeline

PiperOrigin-RevId: 620516628

connor-mccarthy committed:
No public description

PiperOrigin-RevId: 620699930

TheMichaelHu committed:
fix(components): Make AutoSxS autorater_prompt_parameters required

PiperOrigin-RevId: 620855438

Commit:
docs(components): internal

PiperOrigin-RevId: 621215497

Tomcli committed:
chore(components): Update kserve component to v0.12.0 (#10652)

Signed-off-by: tomcli <tommy.chaoping.li@ibm.com>

gmfrasca committed:
fix(backend): Update backend common code and integration tests with updated API Service Params (#10640)

* fix(backend): Refactor backend common code to use updated API Service Params

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

* fix(backend): Fix Backend int tests to use updated API Service Params

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

* WIP: fix(backend): Manually correct the proto schemes to include https

- **NOTE**: this was manually updated, tested, and verified locally
            uploading for CI check
- It appears when regenerating the backend API, something in the
  generation libraries changed and now default the scheme to just
  http, not http+https, which appears to break tests.
- Need to figure out what options to provide api generators to revert
  DefaultScheme to include https again automatically

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

* chore: Update small syntax change in SWF expected test result check

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

* chore: Update backend .proto files to include http and https scheme

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

---------

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

revit13 committed:
fix(Backend + SDK): Add missing optional field to SecretAsVolume and … (#10550)

* fix(Backend + SDK): Add missing optional field to SecretAsVolume and ConfigMapAsVolume.

Signed-off-by: Revital Sur <eres@il.ibm.com>

* Update after rebase.

Signed-off-by: Revital Sur <eres@il.ibm.com>

* Update after rebase.

Signed-off-by: Revital Sur <eres@il.ibm.com>

* Update after merge.

Signed-off-by: Revital Sur <eres@il.ibm.com>

* Updates after merge with master branch.

Signed-off-by: Revital Sur <eres@il.ibm.com>

---------

Signed-off-by: Revital Sur <eres@il.ibm.com>

rawc0der committed:
chore(manifests): refactor manifests for kustomize5 compatibility. Part of #10053 (#10087)

* refactor ml-pipeline component for kustomize5 compatibility

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* rm bk folder

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* fix kustomizations using auto kustomize edit

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* fix env/aws

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* fix kustomize version v5.0.3 in tests

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* minor changes to manifest apply method in pipeline deployment script

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* fix

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* fix kustomize release

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* fix archive

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* fix bin

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* replace patchStrategicMerge refs in test manifests

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* replace kustomize version with 5.2.1

Signed-off-by: Alin Spinu <spinualin@gmail.com>

* replace all kustomize versions with 5.2.1

Signed-off-by: Alin Spinu <spinualin@gmail.com>

---------

Signed-off-by: Alin Spinu <spinualin@gmail.com>

Commit:
fix(components): Fix model eval import error in text generation/classification eval pipeline

PiperOrigin-RevId: 621897220

gmfrasca committed:
fix(frontend): Fix Sidebar tabs to work with argo pod name-id mismatch

- Stemming from upgrade to argo 3.4, Pod Name is no longer always the
  same as NodeID, which breaks a few tabs (PodInfo, PodEvents and
  PodLogs).  Add function to address this

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

Commit:
chore(components): GCPC 2.13.1 Release

PiperOrigin-RevId: 621998414

dependabot[bot] committed:
chore(deps): bump follow-redirects from 1.6.1 to 1.15.6 in /frontend (#10575)

Bumps [follow-redirects](https://github.com/follow-redirects/follow-redirects) from 1.6.1 to 1.15.6.
- [Release notes](https://github.com/follow-redirects/follow-redirects/releases)
- [Commits](https://github.com/follow-redirects/follow-redirects/compare/v1.6.1...v1.15.6)

---
updated-dependencies:
- dependency-name: follow-redirects
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump pillow from 10.0.1 to 10.3.0 in /backend/src/apiserver/visualization (#10658)

Bumps [pillow](https://github.com/python-pillow/Pillow) from 10.0.1 to 10.3.0.
- [Release notes](https://github.com/python-pillow/Pillow/releases)
- [Changelog](https://github.com/python-pillow/Pillow/blob/main/CHANGES.rst)
- [Commits](https://github.com/python-pillow/Pillow/compare/10.0.1...10.3.0)

---
updated-dependencies:
- dependency-name: pillow
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
fix(components): Remove the unused generate_default_instruction and resolve_upload_location from function_based

PiperOrigin-RevId: 622081254

dependabot[bot] committed:
chore(deps): bump express from 4.18.2 to 4.19.2 in /frontend (#10639)

Bumps [express](https://github.com/expressjs/express) from 4.18.2 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.18.2...4.19.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

chensun committed:
chore: Update argo images to 3.4.16 (#10618)

Signed-off-by: Chen Sun <chensun@users.noreply.github.com>

Commit:
feat(components): Use larger base reward model when tuning `text` and `chat` variants of `bison@001` with the `preview.llm.rlhf_pipeline`

PiperOrigin-RevId: 622229648

dependabot[bot] committed:
chore(deps): bump follow-redirects from 1.5.10 to 1.15.6 in /frontend/server (#10574)

Bumps [follow-redirects](https://github.com/follow-redirects/follow-redirects) from 1.5.10 to 1.15.6.
- [Release notes](https://github.com/follow-redirects/follow-redirects/releases)
- [Commits](https://github.com/follow-redirects/follow-redirects/compare/v1.5.10...v1.15.6)

---
updated-dependencies:
- dependency-name: follow-redirects
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

gregsheremeta committed:
chore(sample): add note about secret needing to be pre-created (#10659)

these samples currently only work on the Google Cloud distribution
because of hardcoded secret name that only pre-exists there.
1 extract the hardcoded secret name to a const to make it a little
easier to change
2 add a note about it.

Signed-off-by: Greg Sheremeta <gshereme@redhat.com>

Tomcli committed:
Chore(components): Clean up old ibm components (#10680)

* remove deprecated ibm components

Signed-off-by: tomcli <tommy.chaoping.li@ibm.com>

* update linkage to the ibm components

Signed-off-by: tomcli <tommy.chaoping.li@ibm.com>

---------

Signed-off-by: tomcli <tommy.chaoping.li@ibm.com>

Commit:
chore(components): Update AutoSxS and RLHF image tags

PiperOrigin-RevId: 623200033

abaland committed:
feat(sdk+backend): Add support for generic ephemeral volume (#10605)

* feat(sdk+backend): Add add_ephemeral_volume method to python sdk + add support to backend
Signed-off-by: abaland <abaland@indeed.com>

* feat(sdk+backend): Add add_ephemeral_volume method to python sdk + add support to backend

Signed-off-by: abaland <abaland@indeed.com>

* chore: upgrade go module + go mod tidy

Signed-off-by: abaland <abaland@indeed.com>

* chore: upgrade license files

Signed-off-by: abaland <abaland@indeed.com>

---------

Signed-off-by: abaland <abaland@indeed.com>

Commit:
feat(components): Use larger base reward model when tuning `t5-xxl` with the `preview.llm.rlhf_pipeline`

PiperOrigin-RevId: 623336791

Tomcli committed:
chore(backend): Promote @rimolive as the backend reviewer (#10689)

Signed-off-by: tomcli <tommy.chaoping.li@ibm.com>

gmfrasca committed:
test: update frontend CI to accommodate pod id/name changes

Signed-off-by: Giulio Frasca <gfrasca@redhat.com>

connor-mccarthy committed:
chore(sdk): release `kfp-kubernetes` 1.2.0 (#10692)

Signed-off-by: connor-mccarthy <mccarthy.connor.james@gmail.com>

connor-mccarthy committed:
chore(sdk): make `kfp-kubernetes` release instructions public (#10693)

Signed-off-by: connor-mccarthy <mccarthy.connor.james@gmail.com>

cjidboon94 committed:
chore(kfp-kubernetes): change type of affinity weight to int32 (#10671)

Signed-off-by: Cornelis Boon <cjidboon94@gmail.com>

Commit:
fix(components): Fix image version parameter in rl pipelines

PiperOrigin-RevId: 624348564

Commit:
docs(components): internal

PiperOrigin-RevId: 625114315

Commit:
feat(backend): Merge kfp-tekton backend code (#10678)

* Merge kfp-tekton backend code

Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>

* Add swf work

Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>

---------

Signed-off-by: Ricardo M. Oliveira <rmartine@redhat.com>
